c_src/membrane_ffmpeg_swscale_plugin/scaler.c

```diff
@@ -0,0 +1,200 @@
+#include "scaler.h"
+
+void handle_destroy_state(UnifexEnv *env, State *state) {
+  UNIFEX_UNUSED(env);
+
+  if (state->sws_context != NULL) {
+    sws_freeContext(state->sws_context);
+  }
+
+  for (int i = 0; i < 4; i++) {
+    av_freep(&state->source_data[i]);
+    av_freep(&state->scaled_data[i]);
+    av_freep(&state->target_data[i]);
+  }
```

Comment:
This is not the right way to free that data - `av_image_alloc` creates a continuous memory block and docs suggest using only first pointer https://ffmpeg.org/doxygen/trunk/group__lavu__picture.html#ga841e0a89a642e24141af1918a2c10448